extensions/okhttp/src/main/java/com/google/android/exoplayer2/ext/okhttp/OkHttpDataSource.java

```diff
@@ -280,7 +276,6 @@ private Request makeRequest(DataSpec dataSpec) {
       }
       builder.addHeader("Range", rangeRequest);
     }
-    builder.addHeader("User-Agent", userAgent);
```

Comment:
I'm not sure why you'd want to share an `OkHttpClient` instance and mutate what is essentially its config. This seems like the wrong place to set the `User-Agent` to me. I get that a breaking change could be annoying. I'd rather make it optional than not have any change.